powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java

```diff
@@ -125,4 +459,740 @@ void testUpdateActivation_badRequest() throws Exception {
+     * predefined application ID, template name, user ID, and proximity OTP settings.
+     *
+     * @param proximityOtp a boolean indicating whether proximity OTP is enabled
+     * @return a configured {@link OperationCreateRequest} instance
+     * @throws Exception if the operation creation request setup fails
+     */
+    private OperationCreateRequest createOperationCreateRequest(final boolean proximityOtp) throws Exception {
+        final OperationCreateRequest operationCreateRequest = new OperationCreateRequest();
+        operationCreateRequest.setApplications(List.of(APPLICATION_ID));
+        operationCreateRequest.setTemplateName(TEMPLATE_NAME);
+        operationCreateRequest.setUserId(USER_ID);
+        operationCreateRequest.setProximityCheckEnabled(proximityOtp);
+        return operationCreateRequest;
+    }
+
+    /**
+     * Creates a request object for getting a list of callback URLs.
+     * <p>
+     * This helper method constructs and returns a {@link GetCallbackUrlListRequest} for a
+     * specific application ID.
+     *
+     * @return a configured {@link GetCallbackUrlListRequest} instance
+     */
+    private static GetCallbackUrlListRequest createCallbackUrlListRequest() {
+        final GetCallbackUrlListRequest getCallbackUrlListRequest = new GetCallbackUrlListRequest();
+        getCallbackUrlListRequest.setApplicationId(APPLICATION_ID);
+        return getCallbackUrlListRequest;
+    }
+
+    /**
+     * Creates a request object for creating a callback URL.
+     * <p>
+     * This helper method constructs and returns a {@link CreateCallbackUrlRequest} with
+     * predefined callback URL, name, type, application ID, and other settings.
+     *
+     * @return a configured {@link CreateCallbackUrlRequest} instance
+     */
+    private static CreateCallbackUrlRequest createCallbackUrlRequest() {
+        final CreateCallbackUrlRequest callbackUrlRequest = new CreateCallbackUrlRequest();
+        callbackUrlRequest.setCallbackUrl(CALLBACK_URL);
+        callbackUrlRequest.setName(CALLBACK_NAME);
+        callbackUrlRequest.setType(CallbackUrlType.ACTIVATION_STATUS_CHANGE.name());
+        callbackUrlRequest.setApplicationId(APPLICATION_ID);
+        callbackUrlRequest.setAttributes(Collections.singletonList("activationId"));
+        callbackUrlRequest.setAuthentication(null);
+        return callbackUrlRequest;
+    }
+
+    /**
+     * Wraps an object in a 'requestObject' JSON structure.
+     * <p>
+     * This helper method wraps a given object within a 'requestObject' JSON structure, useful
+     * for preparing request bodies for MockMvc calls.
+     *
+     * @param obj the object to be wrapped in the request JSON
+     * @return a JSON string representation of the object wrapped in 'requestObject'
+     * @throws Exception if JSON processing fails
+     */
+    private static String wrapInRequestObjectJson(final Object obj) throws Exception {
```

Comment:
Personally, I do not like this method at all for several reasons. Mainly, we bypass testing of object mapper. For example, when the property name changes, the test should fail (in my opinion), but in this case it does not. May we use just string payload instead?